ui/pages/send/send-content/send-content.component.js

```diff
@@ -83,24 +91,53 @@ export default class SendContent extends Component {
     );
   }
 
+  checkContractAddress = async () => {
+    const { recipient, tokenAddressList } = this.props;
+
+    const symbolAndDecimals = await getSymbolAndDecimals(
+      recipient.userInput,
+      tokenAddressList,
+    );
+
+    if (
+      (symbolAndDecimals.symbol !== undefined ||
+        symbolAndDecimals.decimals !== undefined) &&
+      symbolAndDecimals.symbol !== ''
+    ) {
+      this.setState({ isKnownContractAddress: true });
+    }
+  };
+
   maybeRenderAddContact() {
     const { t } = this.context;
-    const { isOwnedAccount, contact = {}, to } = this.props;
-    const { showNicknamePopovers } = this.state;
+    const { isOwnedAccount, contact = {}, to, recipient } = this.props;
+    const { showNicknamePopovers, isKnownContractAddress } = this.state;
 
     if (isOwnedAccount || contact.name) {
-      return null;
+      return recipient.warning === 'knownAddressRecipient' ||
+        isKnownContractAddress ? (
```

Comment:
These knownContract addresses are only on Mainnet so should we only show this warning when we're on Ethereum mainnet? cc @danjm

Reply:
No, I think it is okay. Because if we are not on mainnet, then we definitely don't want the user sending to those token addresses. I think improve warnings for that case are being developed, but this actually has benefit in the meantime.